api/v1beta1/types.go

```diff
@@ -733,6 +735,23 @@ func (s SubnetSpec) IsNatGatewayEnabled() bool {
 	return s.NatGateway.Name != ""
 }
 
+// IsIPv6Enabled returns whether or not IPv6 is enabled on the subnet.
+func (s SubnetSpec) IsIPv6Enabled() bool {
+	isIpv6Enabled := false
+	for _, cidr := range s.CIDRBlocks {
```

Comment:
You can use `net.IsIPv6CIDRString(cidr)` instead of the current codeblock to avoid repeated code.

Example here: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/azure/scope/cluster.go#L562

Reply:
Done as suggestion!